File: lib-cli/tmp/server-ts/src/producer/events/serverevents.useralert.ts
import * as Protocol from "implementation/protocol";
import {
	Identification,
	Filter,
	Producer,
	Context,
} from "implementation/events";
import { Output } from "implementation/events/serverevents.useralert";

export function emit(
	event: Protocol.ServerEvents.UserAlert,
	filter: Filter,
	context: Context,
	producer: Producer
): Promise<Output> {
	return Promise.reject(
		new Error(`Handler for event "useralert" isn't implemented`)
	);
}

File: examples/producer/rust/src/producer/events/serverevents_useralert.rs
use super::{identification, producer::Control, protocol, Context, scope::AnonymousScope};
use clibri::server;
use uuid::Uuid;

type BroadcastEventsMessage = (Vec<Uuid>, protocol::Events::Message);
type BroadcastEventsUserConnected = Option<(Vec<Uuid>, protocol::Events::UserConnected)>;

#[allow(unused_variables)]
pub async fn emit<E: server::Error, C: server::Control<E>>(
    event: protocol::ServerEvents::UserAlert,
    scope: &mut AnonymousScope<'_, E, C>,
) -> Result<(BroadcastEventsMessage, BroadcastEventsUserConnected), String> {
    panic!("Handler for protocol::ServerEvents::UserAlert isn't implemented");
}